lib/ecto/type.ex

```diff
@@ -194,6 +194,10 @@ defmodule Ecto.Type do
   @spec type(t) :: t
   def type(type)
 
+  def type({:embed, %{cardinality: :one}}), do: :map
```

Comment:
This will never be called for an embed for Ecto, so we can remove those three clauses.